sktime/forecasting/base/adapters/_pytorchforecasting.py

```diff
@@ -0,0 +1,445 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements adapter for pytorch-forecasting models."""
+import abc
+import functools
+import typing
+from copy import deepcopy
+from typing import Any, Dict, Optional
+
+import numpy as np
+import pandas
+from pandas.api.types import is_numeric_dtype
+
+from sktime.forecasting.base import BaseGlobalForecaster, ForecastingHorizon
+
+__all__ = ["_PytorchForecastingAdapter"]
+__author__ = ["XinyuWu"]
+
+
+class _PytorchForecastingAdapter(BaseGlobalForecaster):
+    """Base adapter class for pytorch-forecasting models.
+
+    Parameters
+    ----------
+    model_params :  Dict[str, Any] (default=None)
+        parameters to be passed to initialize the underlying pytorch-forecasting model
+        for example: {"lstm_layers": 3, "hidden_continuous_size": 10} for TFT model
+    dataset_params : Dict[str, Any] (default=None)
+        parameters to initialize `TimeSeriesDataSet` [1]_ from `pandas.DataFrame`
+        max_prediction_length will be overwrite according to fh
+        time_idx, target, group_ids, time_varying_known_reals, time_varying_unknown_reals
+        will be infered from data, so you do not have to pass them
+    train_to_dataloader_params : Dict[str, Any] (default=None)
+        parameters to be passed for `TimeSeriesDataSet.to_dataloader()`
+        by default {"train": True}
+    validation_to_dataloader_params : Dict[str, Any] (default=None)
+        parameters to be passed for `TimeSeriesDataSet.to_dataloader()`
+        by default {"train": False}
+
+    References
+    ----------
+    .. [1] https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.data.timeseries.TimeSeriesDataSet.html
+    """  # noqa: E501
+
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": ["XinyuWu"],
+        "maintainers": ["XinyuWu"],
+        "python_version": ">=3.8",
+        "python_dependencies": ["pytorch_forecasting"],
```

Comment:
you have set version bounds in the specific classes but not here. Can you explain this?

Reply:
No specific reason here. When I first started this PR, I was trying to learn from the exsiting neuralforecast interface. Yeah, I will set version bounds in both classes.